src/Atmos/Model/precipitation.jl

```diff
@@ -100,39 +147,104 @@ function compute_gradient_flux!(
     aux::Vars,
     t::Real,
 )
-    # diffusive flux of q_tot
     diffusive.precipitation.∇q_rai = ∇transform.precipitation.q_rai
 end
 
+function flux_first_order!(
+    precip::RainModel,
+    atmos::AtmosModel,
+    flux::Grad,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+    ts,
+    direction,
+)
+    tend = Flux{FirstOrder}()
+    args = (atmos, state, aux, t, ts, direction)
+    flux.precipitation.ρq_rai = Σfluxes(eq_tends(Rain(), atmos, tend), args...)
+end
+
+function flux_second_order!(
+    precip::RainModel,
+    flux::Grad,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    D_t,
+)
+    d_q_rai = (-D_t) .* diffusive.precipitation.∇q_rai
+
+    flux_second_order!(precip, flux, state, d_q_rai)
+end
+function flux_second_order!(precip::RainModel, flux::Grad, state::Vars, d_q_rai)
+    flux.precipitation.ρq_rai += d_q_rai * state.ρ
+end
```

Comment:
We could use the new tendency spec for this, but I'm fine leaving that for a separate PR.

Reply:
Sounds good. Lets do it in the new one. This is already too big PR if you ask me. Who wrote it? ;)